### DISCUSSION TITLE
Mastodon su superuser password lost -- any way to reset from root or am I stuffed?

### DISCUSSION THREAD

**Discussion by lookitmychicken** [mastodon/mastodon#20364] [Q&A]
I did baby's first mastodon install over the last couple of days, and copied the superuser password into a text file. But then I didn't save the freaking text file before I had a computer hang. Forgot it was unsaved, hard reset computer, cannot recover superuser password I think.

Is there any recourse for me? Or do I just need to shrug, accept the learning opportunity, wipe & restart?

**Comment by MrGeneration** [comment]
Using the password reset function doesn't work...?

![image](https://user-images.githubusercontent.com/6549061/201220385-d0be8391-e105-4831-babe-d9bf22184b15.png)

You can find that form via `https://your.instance/auth/password/new`

![image](https://user-images.githubusercontent.com/6549061/201220416-3b38a4cb-6067-4b63-92e2-ee18511295fa.png)

**Reply by lookitmychicken** [reply]
Alas no, I'm trying to reset the root password for the mastodon superuser from the Ubuntu instance hosting the mastodon server. Not the admin password :(

**Reply by lookitmychicken** [reply]
(I thought it was the same thing, but apparently not)

**Reply by MrGeneration** [reply]
errr... if you have no root access to your server... you're... well, kinda doomed.

**Reply by lookitmychicken** [reply]
I had root access to the server, just not mastodon superuser access after I'd logged in as root. So, whoopsie! It's all a learning experience, I've started yet another fresh install. 😂

**Reply by rbairwell** [reply]
Just to clarify: You CAN SSH into the machine as "root" but you can't login as on the command line as "mastodon"?

If so, `su - mastodon` will switch you to the Mastodon user and `passwd mastodon` will (if run as root) will allow you to change the Mastodon user account's password.

**Reply by crecente** [reply]
I had a similar problem when moving our instance to a new server.

I was able to switch back to the root user and then `passwd root` to set the root account's password. 

(It might be that this only worked because the new server did not have a password for root. I don't know but thought I'd share that anyway as it helped me in my situation.)
